holoviews/plotting/bokeh/callbacks.py

```diff
@@ -1,446 +1,225 @@
-           kernel.execute(pyimport + cmd, callbacks, {{silent : false}});
-        }}
-
-        if (!HoloViewsWidget._queued) {{
-            HoloViewsWidget._queued = [];
-            HoloViewsWidget._blocked = false;
-            HoloViewsWidget._timeout = Date.now();
-        }}
-
-        timeout = HoloViewsWidget._timeout + {timeout};
-        if ((typeof _ === "undefined")  || _.isEmpty(data)) {{
-        }} else if ((HoloViewsWidget._blocked && (Date.now() < timeout))) {{
-            HoloViewsWidget._queued = [data];
-        }} else {{
-            HoloViewsWidget._queued = [data];
-            setTimeout(execute_callback, {delay});
-            HoloViewsWidget._blocked = true;
-            HoloViewsWidget._timeout = Date.now();
+    attributes = {}
+
+    js_callback = """
+        var argstring = JSON.stringify(data);
+        if ((window.Jupyter !== undefined) && (Jupyter.notebook.kernel !== undefined)) {{
+          var comm_manager = Jupyter.notebook.kernel.comm_manager;
+          var comms = HoloViewsWidget.comms["{comms_target}"];
+          if (comms && ("{comms_target}" in comms)) {{
+            comm = comms["{comms_target}"];
+          }} else {{
+            comm = comm_manager.new_comm("{comms_target}", {{}}, {{}}, {{}});
+
+          comm_manager["{comms_target}"] = comm;
+            HoloViewsWidget.comms["{comms_target}"] = comm;
+          }}
+          comm_manager["{comms_target}"] = comm;
+          comm.send(argstring)
         }}
     """
 
-    def initialize(self, data):
-        """
-        Initialize is called when the callback is added to a new plot
-        and the initialize option is enabled. May avoid repeat
-        initialization by setting initialize_cb parameter to False
-        inside this method.
-        """
-
-
-    def __call__(self, data):
-        """
-        The call method can modify any bokeh plot object
-        depending on the supplied data dictionary. It
-        should return the modified plot objects as a list.
-        """
-        return []
-
-
-    def update(self, data, chained=False):
-        """
-        The update method is called by the javascript callback
-        with the supplied data and will return the json serialized
-        string representation of the changes to the Bokeh plot.
-        When chained=True it will return a list of the plot objects
-        to be updated, allowing chaining of callback operations.
-        """
-        if self.skip_unchanged and self.current_data == data:
-            return [] if chained else "{}"
-        self.current_data = data
-
-        objects = self(data)
-
+    # The plotting handle(s) to attach the JS callback on
+    handles = []
+    msgs=[]
+
+    def __init__(self, plot, streams, source, **params):
+        self.plot = plot
+        self.streams = streams
+        self.comm = JupyterCommJS(plot, on_msg=self.on_msg)
+        self.source = source
+
+
+    def initialize(self):
```

Comment:
If I understand this right, the docstring could say something like 'Iterates through the plot (and any subplots), registering the necessary JS code across all relevant plotting handles'. Again, that is what I expect it to be doing from the code!